lib/pbench/test/unit/server/test_cache_manager.py

```diff
@@ -439,11 +440,96 @@ def mock_run(args, **kwargs):
             m.setattr(subprocess, "run", mock_run)
             m.setattr(Tarball, "__init__", TestCacheManager.MockTarball.__init__)
             m.setattr(Controller, "__init__", TestCacheManager.MockController.__init__)
+            m.setattr(os.path, "getmtime", lambda self: 1681814586.1138802)
             tb = Tarball(tar, Controller(Path("/mock/archive"), cache, None))
             tb.unpack()
             assert call == ["tar", "find"]
             assert tb.unpacked == cache / "ABC" / tb.name
 
+    def test_cachemap_modification_time(self, monkeypatch):
+        """Test to build the cache map of the unpacked tarball"""
+        tar = Path("/mock/A.tar.xz")
+        cache = Path("/mock/.cache")
+        call = list()
+
+        def mock_run(args, **kwargs):
+            call.append(args[0])
+
+            tar_target = "--file=" + str(tar)
+            assert args[0] == "find" or args[0] == "tar" and tar_target in args
+            return subprocess.CompletedProcess(
+                args, returncode=0, stdout="Successfully Unpacked!", stderr=None
+            )
+
+        def mock_getmtime(path):
+            """Mock file/directory modification time"""
+            if path == Path("/mock/.cache/ABC/A"):
+                raise FileNotFoundError(
+                    f"[Errno 2] No such file or directory: '{path}'"
+                )
+
+        with monkeypatch.context() as m:
+            m.setattr(Path, "mkdir", lambda path, parents: None)
+            m.setattr(subprocess, "run", mock_run)
+            m.setattr(Tarball, "__init__", TestCacheManager.MockTarball.__init__)
+            m.setattr(Controller, "__init__", TestCacheManager.MockController.__init__)
+            m.setattr(os.path, "getmtime", mock_getmtime)
+            tb = Tarball(tar, Controller(Path("/mock/archive"), cache, None))
+            with pytest.raises(FileNotFoundError) as exc:
+                tb.unpack()
+            assert (
+                str(exc.value)
+                == "[Errno 2] No such file or directory: '/mock/.cache/ABC/A'"
+            )
+            assert call == ["tar", "find"]
+            assert tb.unpacked == cache / "ABC" / tb.name
+
+    def test_cachemap_success(self, monkeypatch):
+        """Test to build the cache map of the unpacked tarball"""
+        tar = Path("/mock/A.tar.xz")
+        cache = Path("/mock/.cache")
+        call = list()
+
+        def mock_run(args, **kwargs):
+            call.append(args[0])
+
+            tar_target = "--file=" + str(tar)
+            assert args[0] == "find" or args[0] == "tar" and tar_target in args
+            return subprocess.CompletedProcess(
+                args, returncode=0, stdout="Successfully Unpacked!", stderr=None
+            )
+
+        def mock_glob(self, path):
```

Comment:
This would be a great place for the mock `Path` that Peter is certain he built (and I remember him enthusing about it at the time) but wasn't able to locate. Huh. In any case, it might be easier to read if you built a hierarchy directory structure first and had the mocked `glob` read from that. Maybe.